File: aquila-frontend/src/lib/api/simulation.ts
import { CounterfactualScenario, CounterfactualResult } from './types';
import { apiClient } from './client';

export const simulationApi = {
  async runCounterfactual(scenario: CounterfactualScenario): Promise<CounterfactualResult> {
    return apiClient.post<CounterfactualResult>('/simulation/counterfactual', scenario);
  }
};

File: backend/app/api/v1/simulation.py
from fastapi import APIRouter

from app.schemas.simulation import CounterfactualScenario, CounterfactualResult
from app.services.simulation_service import CounterfactualSimulationService

router = APIRouter()
service = CounterfactualSimulationService()


@router.post("/counterfactual", response_model=CounterfactualResult)
async def run_counterfactual(scenario: CounterfactualScenario):
    """
    Runs a forward counterfactual simulation for a candidate vessel release hypothesis.
    """
    return service.run_scenario(scenario)
